service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/SpecimenMapper.java

```diff
@@ -0,0 +1,239 @@
+        List<Observation> observationsAssociatedWithSpecimen = observations.stream()
+            .filter(Observation::hasSpecimen)
+            .filter(observation -> observation.getSpecimen().getReference().equals(specimen.getId()))
+            .collect(Collectors.toList());
+
+        return observationsAssociatedWithSpecimen.stream().map(
+            observationAssociatedWithSpecimen -> observationMapper.mapObservationToCompoundStatement(
+                observationAssociatedWithSpecimen,
+                observations
+            ))
+            .collect(Collectors.joining());
+    }
+
+    private Optional<String> buildPertinentInformation(Specimen specimen) {
+        PertinentInformationBuilder pertinentInformationBuilder = new PertinentInformationBuilder();
+        if (specimen.hasCollection()) {
+            Specimen.SpecimenCollectionComponent collection = specimen.getCollection();
+
+            collection.getExtensionsByUrl(FASTING_STATUS_URL).stream().findFirst()
+                .ifPresent(extension -> {
+                    Type value = extension.getValue();
+                    if (value instanceof CodeableConcept) {
+                        pertinentInformationBuilder.fastingStatus((CodeableConcept) value);
+                    } else if (value instanceof Duration) {
+                        pertinentInformationBuilder.fastingDuration((Duration) value);
+                    }
+                });
+
+            if (collection.hasQuantity()) {
+                pertinentInformationBuilder.quantity(collection.getQuantity());
+            }
+
+            if (collection.hasBodySite()) {
+                pertinentInformationBuilder.collectionSite(collection.getBodySite());
+            }
+        }
+
+        specimen.getNote().stream()
+            .map(Annotation::getText)
+            .forEach(pertinentInformationBuilder::note);
+
+        return getEffectiveTime(specimen)
+            .map(pertinentInformationBuilder::build)
+            .orElseGet(pertinentInformationBuilder::build);
+    }
+
+    private static class PertinentInformationBuilder {
+
+        private static final String FASTING_STATUS = "Fasting Status:";
+        private static final String FASTING_DURATION = "Fasting Duration:";
+        private static final String QUANTITY = "Quantity:";
+        private static final String COLLECTION_SITE = "Collection Site:";
+        private static final String COMMENT_PREFIX = "comment type - LAB SPECIMEN COMMENT(E271)";
+
+        private String pertinentInformation;
+
+        PertinentInformationBuilder() {
+            pertinentInformation = StringUtils.EMPTY;
+        }
+
+        public void fastingStatus(CodeableConcept fastingStatus) {
+            CodeableConceptMappingUtils.extractTextOrCoding(fastingStatus)
+                .map(fastingStatusValue -> newLine(withSpace(FASTING_STATUS, fastingStatusValue), pertinentInformation))
+                .ifPresent(text -> pertinentInformation = text);
```

Comment:
Perhaps a private `.prepend` method might make it clearer what's going on here? At first glance I thought this was overwriting the value, assuming it was being called first perhaps.

Reply:
Updated in e125b8e7d26d414027105686f6c163cfe8885b53